hazenlib/HazenTask.py

```diff
@@ -10,21 +10,34 @@
 
 class HazenTask:
 
-    def __init__(self, data_paths: list, report: bool = False, report_dir: str = os.path.join(os.getcwd(), 'report')):
-        self.data_paths = sorted(data_paths)
+    def __init__(self, input_data, report: bool = False, report_dir = None):
+        # Check if input data is a single or list of files, load accordingly
+        if isinstance(input_data, list):
+            data_paths = sorted(input_data)
+            self.dcm_list = [dcmread(dicom)for dicom in data_paths]
+        else:
+            self.single_dcm = dcmread(input_data)
         self.report: bool = report
-        self.report_path = os.path.join(report_dir, type(self).__name__)
+        if report_dir is not None:
+            self.report_path = os.path.join(str(report_dir), type(self).__name__)
+        else:
+            self.report_path = os.path.join(os.getcwd(), 'report_image',
+                                            type(self).__name__)
         if report:
             pathlib.Path(self.report_path).mkdir(parents=True, exist_ok=True)
         else:
             pass
         self.report_files = []
 
-    @property
-    def data(self) -> list:
-        return [dcmread(dicom)for dicom in self.data_paths]
+    def init_result_dict(self) -> dict:
+        result_dict = {
+            "task": f"{type(self).__name__}",
+            "file": None,
+            "measurement": {}
+        }
+        return result_dict
 
-    def key(self, dcm, properties=None) -> str:
+    def img_desc(self, dcm, properties=None) -> str:
```

Comment:
rename function to better describe its purpose, it is no longer being used as a dictionary key